File: rules/ux.py
import frappe
from frappe import msgprint, throw, _


# ruleid: frappe-missing-translate-function-python
throw("Error Occured")

# ruleid: frappe-missing-translate-function-python
frappe.throw("Error Occured")

# ruleid: frappe-missing-translate-function-python
frappe.msgprint("Useful message")

# ruleid: frappe-missing-translate-function-python
msgprint("Useful message")


# ok: frappe-missing-translate-function-python
translatedmessage = _("Hello")

# ok: frappe-missing-translate-function-python
throw(translatedmessage)

# ok: frappe-missing-translate-function-python
msgprint(translatedmessage)

# ok: frappe-missing-translate-function-python
msgprint(_("Helpful message"))

# ok: frappe-missing-translate-function-python
frappe.throw(_("Error occured"))

not_translated = "abc"

# ruleid: frappe-missing-translate-function-python
frappe.throw("Some Text" + not_translated + "More text")

# ok: frappe-missing-translate-function-python
frappe.throw(_("Some Text") + not_translated)

# ok: frappe-missing-translate-function-python
frappe.throw(_("Some Text") + not_translated, title=_("None"))

# ruleid: frappe-missing-translate-function-python
frappe.throw(_("Some Text"), title="Warning")

# ok: frappe-missing-translate-function-python
frappe.throw(_("Some Text"), title=_("Warning"))

File: rules/ux.js

// ok: frappe-missing-translate-function-js
frappe.msgprint('{{ _("Both login and password required") }}');

// ruleid: frappe-missing-translate-function-js
frappe.msgprint('What');

// ok: frappe-missing-translate-function-js
frappe.show_alert(__('What'));

// ruleid: frappe-missing-translate-function-js
frappe.show_alert('What');

// ok: frappe-missing-translate-function-js
frappe.throw('  {{ _("Both login and password required") }}.  ');


// ruleid: frappe-missing-translation-button-text
frm.add_custom_button("Transfer Asset", function() {

})

// ok: frappe-missing-translation-button-text
frm.add_custom_button(__("Transfer Asset"), function() {

})

